lib/util/probe.js

```diff
@@ -60,8 +60,39 @@ function observeKafkaStats(msg) {
     kafkaMetrics.observe(JSON.parse(msg.message));
 }
 
+/**
+ * Get probe config will pull the configuration for the probe server based on
+ * the provided site name. If siteNames is empty, it returns the global probe server config
+ * only if it's a single object.
+ *
+ * @param {Object} queueProcessorConfig - Configuration of the queue processor that
+ *      holds the probe server configs for all sites
+ * @param {Array<String>} siteNames - List of site names (should contain at most one element)
+ * @returns {Object|undefined} Config for site or global config, undefined if no match found or invalid config
+ */
+function getProbeConfig(queueProcessorConfig, siteNames) {
+    if (siteNames.length === 0) {
+        if (!Array.isArray(queueProcessorConfig.probeServer)) {
+            return queueProcessorConfig.probeServer;
+        }
+        return undefined;
+    }
+
+    if (Array.isArray(queueProcessorConfig.probeServer)) {
+        if (siteNames.length !== 1) {
+            return undefined;
```

Comment:
returning `undefined` here will generate a strange error : "configuration for probe server is missing" (when starting probe server with undefined probe)

We may want to add an error log here to point to the issue...